src-cpp/rdkafkacpp.h

```diff
@@ -2357,6 +2357,22 @@ class RD_EXPORT KafkaConsumer : public virtual Handle {
    *         RdKafka::KafkaConsumer::assign() */
   virtual ErrorCode assignment (std::vector<RdKafka::TopicPartition*> &partitions) = 0;
 
+
+  /** @brief Check whether the consumer considers the current assignment to
+   *         have been lost involuntarily. This method is only applicable for
+   *         use with a subscribing consumer. Assignments are revoked
+   *         immediately when determined to have been lost, so this method is
+   *         only useful within a rebalance callback.
+   *
+   * @remark Calling rd_kafka_assign(), rd_kafka_incremental_assign() or
+   *         rd_kafka_incremental_unassign() resets this flag.
+   *
+   * @returns Returns true if the current partition assignment is considered
+   *          lost, false otherwise.
+   */
+  virtual bool assignment_lost () = 0;
```

Comment:
New methods must be added to the end of the virtual class to not break ABI compat.

Reply:
so many rules!